code/obj/critter/misc.dm

```diff
@@ -1554,6 +1555,40 @@
 				src.task = "chasing"
 				break
 
+	attackby(obj/item/W, mob/M)
+		if(istype(W, /obj/item/reagent_containers/food/snacks) && !(M in src.friends))
+			if(prob(25))
+				src.visible_message("<span class='notice'>[src] munches happily on the [W], and seems a little friendlier with [M]!</span>")
+				friends += M
+				src.task = "thinking"
+			else
+				src.visible_message("<span class='notice'>[src] hated the [W]! It bit [M]'s hand!</span>")
+				M.reagents?.add_reagent("viper_venom", rand(15,30))
+				playsound(src.loc, 'sound/impact_sounds/Generic_Stab_1.ogg', 50, 1)
+				M.emote("scream")
+			M.drop_item()
+			qdel(W)
+			src.health = min(src.maxhealth, src.health + health_gain_from_food)
+			eat_twitch(src)
+		else
+			..()
+
+	attack_hand(mob/M)
+		if ((M.a_intent != INTENT_HARM) && (M in src.friends))
+			if(M.a_intent == INTENT_HELP && src.aggressive)
+				src.visible_message("<span class='notice'>[M] pats [src] on the head in a soothing way. It won't attack anyone now.</span>")
+				src.aggressive = FALSE
+				src.mobile = 1
+				icon_state = "rattlesnake"
+				src.task = "thinking"
+				return
+			else if((M.a_intent == INTENT_DISARM || M.a_intent == INTENT_GRAB) && !src.aggressive)
```

Comment:
does this mean that punching a tamed rattlesnake won't make it mad, but disarming or grabbing it will

Reply:
it means you can make it aggressive towards other people like it is with scorpions

Reply:
Ah gotcha